Sources/FluentMongoDriver/MongoDB+Join.swift

```diff
@@ -95,8 +95,23 @@ extension DatabaseQuery {
                         "$unwind": "$\(alias ?? schema)"
                     ]))
                 case .custom:
-                    fatalError()
+                    throw FluentMongoError.unsupportedJoin
                 }
+            case .advancedJoin(let schema, let space, let alias, let method, let filters) where space == nil && filters.count == 1:
```

Comment:
One nit: We could replace `let space` with `nil`, removing the need of the `space == nil` clause

Reply:
Fixed 👍